src/librbd/io/ImageRequestWQ.cc

```diff
@@ -642,9 +642,9 @@ bool ImageRequestWQ<I>::needs_throttle(ImageDispatchSpec<I> *item) {
     throttle = t.second;
     tokens = item->tokens_requested(flag);
 
-    if (throttle->get<ImageRequestWQ<I>, ImageDispatchSpec<I>,
-	      &ImageRequestWQ<I>::handle_throttle_ready>(
-		tokens, this, item, flag)) {
+    if (tokens && throttle->get<ImageRequestWQ<I>, ImageDispatchSpec<I>,
+	            &ImageRequestWQ<I>::handle_throttle_ready>(
```

Comment:
Nit: hard to follow the w/ this indentation. Perhaps leave ```tokens &&``` on its own line for clarity